nupic/research/frameworks/dendrites/modules/dendritic_mlp.py

```diff
@@ -166,9 +189,9 @@ def _init_sparse_dendrites(m, input_sparsity):
         # Assume `m` is an instance of `DendriticLayerBase`
         input_density = 1.0 - input_sparsity
         weight_density = 1.0 - m.segments.sparsity
-        fan_in = m.segments.dim_context
+        fan_in = m.dim_context
         bound = 1.0 / np.sqrt(input_density * weight_density * fan_in)
-        nn.init.uniform_(m.segments.weights, -bound, bound)
+        nn.init.uniform_(m.weights, -bound, bound)
```

Comment:
This will have to change based on what modification is made to the `weights` property. I'm just setting a reminder comment.